src/com/owncloud/android/ui/activity/ReceiveExternalFilesActivity.java

```diff
@@ -647,14 +651,77 @@ public boolean onOptionsItemSelected(MenuItem item) {
                     onBackPressed();
                 }
                 break;
-
+            case R.id.action_sort_descending:
+                item.setChecked(!item.isChecked());
+                boolean isAscending = !item.isChecked();
+                com.owncloud.android.db.PreferenceManager.setSortAscending(isAscending,this,FileStorageUtils.FILE_DISPLAY_SORT);
+                switch(com.owncloud.android.db.PreferenceManager.getSortOrder(this,FileStorageUtils.FILE_DISPLAY_SORT)) {
+                    case FileStorageUtils.SORT_NAME:
+                        sortByName(isAscending);
+                        break;
+                    case FileStorageUtils.SORT_SIZE:
+                        sortBySize(isAscending);
+                        break;
+                    case FileStorageUtils.SORT_DATE:
+                        sortByDate(isAscending);
+                        break;
+                }
+                break;
+            case R.id.action_sort_by_name:
+                item.setChecked(true);
+                sortByName(com.owncloud.android.db.PreferenceManager.getSortAscending(this,FileStorageUtils.FILE_DISPLAY_SORT));
+                break;
+            case R.id.action_sort_by_size:
+                item.setChecked(true);
+                sortBySize(com.owncloud.android.db.PreferenceManager.getSortAscending(this,FileStorageUtils.FILE_DISPLAY_SORT));
+                break;
+            case R.id.action_sort_by_date:
+                item.setChecked(true);
+                sortByDate(com.owncloud.android.db.PreferenceManager.getSortAscending(this,FileStorageUtils.FILE_DISPLAY_SORT));
```

Comment:
I'm curious about this, do we need another case to save this preference? Something like FILE_SHARE_SORT?

Reply:
@davigonz I think it would be better if the sort order in while sharing files would be same as that while uploading files.